agents/text_2_sql_bedrock_agent/dependencies/clean.py

```diff
@@ -0,0 +1,254 @@
+    try:
+        glue_client.delete_database(Name=database_name)
+        print(f"Database '{database_name}' deleted successfully.")
+    except Exception as e:
+        print(f"Error deleting database '{database_name}':", e)
+
+
+delete_crawler(glue_crawler_name)
+delete_tables(glue_database_name)
+delete_database(glue_database_name)
+
+
+
+#agent_name = "text-2-sql-agent"
+
+
+
+
+
+list_agent=bedrock_agent_client.list_agents()['agentSummaries']
+list_agent
+# Search for the agent with the name 'text2sql' and extract its ID
+agent_id = next((agent['agentId'] for agent in list_agent if agent['agentName'] == agent_name), None)
+
+print(agent_id)
+
+response = bedrock_agent_client.list_agent_action_groups(
+    agentId=agent_id,
+    agentVersion='1',
+
+)
+list_action_group=response['actionGroupSummaries']
+print(list_action_group)
+
+action_group_name='QueryAthenaActionGroup'
+
+action_group_id=next((agent['actionGroupId'] for agent in list_action_group if agent['actionGroupName'] == action_group_name), None)
+print(action_group_id)
+
+response = bedrock_agent_client.list_agent_aliases(
+    agentId=agent_id,
+)
+response['agentAliasSummaries']
+print(type(response['agentAliasSummaries']))
+agentAliasId=next((agent['agentAliasId'] for agent in response['agentAliasSummaries'] if agent['agentAliasName'] == agent_alias_name), None)
+agentAliasId
+
+
+agent_name = "text-2-sql-agent"
+
+lambda_name = f'{agent_name}-{suffix}'
+print(lambda_name)
+
+resp=lambda_client.get_function(FunctionName=lambda_name)
+print(resp['Configuration']['FunctionArn'])
+FunctionArn=resp['Configuration']['FunctionArn']
+try:
+    response = bedrock_agent_client.update_agent_action_group(
+       agentId=agent_id,
+       agentVersion='DRAFT',
+       actionGroupId= action_group_id,
+       actionGroupName=action_group_name,
+       actionGroupExecutor={
+           'lambda': FunctionArn
+       },
+       apiSchema={
+           's3': {
+               's3BucketName': bucket_name,
+               's3ObjectKey': bucket_key
+           }
+       },
+       actionGroupState='DISABLED',
+    )
+
+
+    action_group_deletion = bedrock_agent_client.delete_agent_action_group(
+       agentId=agent_id,
+       agentVersion='DRAFT',
+       actionGroupId= action_group_id
+    )
+except:
+    print('can not delete')
+
+
+agent_alias_deletion = bedrock_agent_client.delete_agent_alias(
+   agentId=agent_id,
+   agentAliasId=agentAliasId
+)
+
+
+agent_deletion = bedrock_agent_client.delete_agent(
+   agentId=agent_id
+)
+
+
+
+
+
+# Delete Lambda function
+lambda_client.delete_function(
+    FunctionName=lambda_name
+)
+
+
+
+
+
+# Empty and delete S3 Bucket
+
+objects = s3_client.list_objects(Bucket=bucket_name)  
+if 'Contents' in objects:
+    for obj in objects['Contents']:
+        s3_client.delete_object(Bucket=bucket_name, Key=obj['Key']) 
+s3_client.delete_bucket(Bucket=bucket_name)
+
+
+
+
+
+policy_arns = [
+    'arn:aws:iam::aws:policy/AmazonAthenaFullAccess',
```

Comment:
can those be less granular?

Reply:
I will check, everything else has been addressed, except these two Athena and Glue full access